### PR TITLE
feat: add `array/base/unary3d-by`

### LINKED ISSUE #3173: [RFC]: add `array/base/unary3d-by`
### Description

This RFC proposes adding a new package to the `array/base/*` namespace: `@stdlib/array/base/unary3d-by`.

This package is the 3d equivalent of [`@stdlib/array/base/unary2d-by`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/unary2d-by).

To see an example of a similar package, but one which doesn't support accessors, see [`@stdlib/array/base/unary3d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/unary3d).

While the package proposed in this RFC will **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by Neerajpathak07**
Resolves #3173  
## Description

> What is the purpose of this pull request?

This pull request:

-   adds implementation for `array/base/unary3d-by`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3173 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@Neerajpathak07 CI failures need to be addressed before this PR can be reviewed.

**Comment by Neerajpathak07** [comment]
@kgryte Regarding the run_affected_examples they are running perfectly on my system when I tested them locally:-
unary3d-by

**Comment by Neerajpathak07** [comment]
I'll resolve the lint errors and commit again.

File: lib/node_modules/@stdlib/array/base/unary3d-by/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+var shape = [ 3, 3, 3 ];
+
+var x = filled3dBy( shape, discreteUniform( -100, 100 ) );
+console.log( x );
+
+var y = zeros3d( shape );
+console.log( y );
+
+unary23By( [ x, y ], shape, abs, accessor );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/unary3d-by/README.md]:
```suggestion
unary3dBy( [ x, y ], shape, abs, accessor );
```

File: lib/node_modules/@stdlib/array/base/unary3d-by/test/test.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+tape( 'the function applies a function to each nested input array element according to a callback function and assigns results to a nested output array', function test( t ) {
+	var expected;
+	var shape;
+	var x;
+	var y;
+
+	shape = [ 2, 2, 2 ];
+	x = [
+		    [
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/unary3d-by/test/test.js]: JavaScript files should always use TAB indentation. Please ensure you have properly configured EditorConfig, per the development guide.

Comment (Neerajpathak07) [reply lib/node_modules/@stdlib/array/base/unary3d-by/test/test.js]: Yea, I'll look into it!

**Comment by Neerajpathak07** [comment]
@kgryte all CI checks are passing!!

File: lib/node_modules/@stdlib/array/base/unary3d-by/docs/types/test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	unary3dBy( [ x, y ], {}, identity, identity, {} ); // $ExpectError
+	unary3dBy( [ x, y ], ( x: number ): number => x, identity, identity, {} ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a third argument which is not a valid function...
+{
+	const x = [ [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ] ];
+	const y = [ [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ] ];
+
+	unary3dBy( [ x, y ], [ 1, 2,1,  2 ], 'abc', identity ); // $ExpectError
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/base/unary3d-by/docs/types/test.ts]:
```suggestion
	unary3dBy( [ x, y ], [ 1, 2, 1, 2 ], 'abc', identity ); // $ExpectError
```

File: lib/node_modules/@stdlib/array/base/unary3d-by/package.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    "type": "git",
+    "url": "git://github.com/stdlib-js/stdlib.git"
+  },
+  "bugs": {
+    "url": "https://github.com/stdlib-js/stdlib/issues"
+  },
+  "dependencies": {},
+  "devDependencies": {
+      "eslint": "9.15.0"
+  },
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/unary3d-by/package.json]:
```suggestion
  "devDependencies": {},
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Neerajpathak07!

**Pull request merged**